ql/src/java/org/apache/hadoop/hive/ql/HiveQueryLifeTimeHook.java

```diff
@@ -75,13 +75,15 @@ private void checkAndRollbackCTAS(QueryLifeTimeHookContext ctx) {
     PrivateHookContext pCtx = (PrivateHookContext) ctx.getHookContext();
     Path tblPath = pCtx.getContext().getLocation();
 
-    try {
-      FileSystem fs = tblPath.getFileSystem(conf);
-      if (!fs.exists(tblPath)) {
-        return;
+    if (tblPath != null) {
+      try {
+        FileSystem fs = tblPath.getFileSystem(conf);
+        if (!fs.exists(tblPath)) {
+          return;
+        }
+      } catch (Exception e) {
+        throw new RuntimeException("Not able to check whether the CTAS table directory exists due to: ", e);
```

Comment:
I am not sure if the error message is reasonable, as some failed queries may not be `CTAS ` type. Welcome to come up with comment here.

Reply:
The method calling is ```private void checkAndRollbackCTAS(QueryLifeTimeHookContext ctx) {``` and you mean to say this method gets invoked for non CTAS queries as well?

Reply:
> you mean to say this method gets invoked for non CTAS queries as well?

Yse, In my case HIVE-27565 which is `Drop` type , also invoked `checkAndRollbackCTAS(QueryLifeTimeHookContext ctx)`. But i think this method is only for handling `CTAS` type queries, so other queries should return early to leave the method.

I will refine the PR to return early if the query is not `CTAS`.